packages/lib/storage/src/new/read.ts

```diff
@@ -0,0 +1,224 @@
+        conversationManifestKey,
+        db,
+    );
+
+    if (!conversationManifest) {
+        throw Error(`Conversation manifest not found`);
+    }
+
+    return conversationManifest;
+}
+
+/**
+ * This function retrieves the manifest for the account.
+ *
+ * @param {Db} db - The database object, which contains the account's ENS name and the sign function.
+ * @returns {Promise<AccountManifest>} A promise that resolves to the account manifest.
+ * @throws {Error} If the account manifest does not exist.
+ */
+export async function getAccountManifest(db: Db): Promise<AccountManifest> {
+    const accountManifestKey = await getAccountManifestKey(db);
+    const accountMainfest = await readFromStrorage<AccountManifest>(
+        accountManifestKey,
+        db,
+    );
+
+    if (!accountMainfest) {
+        throw Error(`Account manifest not found`);
+    }
+
+    return accountMainfest;
+}
+
+/**
+ * This function retrieves the total number of conversations associated with an account.
+ *
+ * @param {Db} db - The database object, which contains the account's ENS name and the sign function.
+ * @returns {Promise<number>} A promise that resolves to the total number of conversations.
+ * @throws {Error} If the account manifest does not exist.
+ */
+export async function getNumberOfConverations(db: Db): Promise<number> {
+    const accountManifestKey = await getAccountManifestKey(db);
+    const accountMainfest = await readFromStrorage<AccountManifest>(
+        accountManifestKey,
+        db,
+    );
+
+    if (!accountMainfest) {
+        throw Error(`Account manifest not found`);
+    }
+
+    return accountMainfest.conversationListCounter;
+}
+
+async function localFirstRead<T extends Chunk>(
+    key: string,
+    db: Db,
+    remoteFetchCb: RemoteFetchCb | undefined,
+) {
+    const localChunk = await db.keyValueStoreLocal.read(key);
+
+    if (localChunk) {
+        // Return the local chunk if it exists
+        return localChunk as T;
+    }
+
+    if (db.keyValueStoreRemote) {
+        // Only try to read from remote if the local chunk does not exist
+        const remoteChunk: T | undefined = await db.keyValueStoreRemote.read(
+            key,
+        );
+
+        if (remoteChunk) {
+            if (remoteFetchCb) {
+                remoteFetchCb(key, remoteChunk);
+            }
+
+            return remoteChunk;
+        }
+    }
+
+    return undefined;
+}
+
+async function remoteFirstRead<T extends Chunk>(
+    key: string,
+    db: Db,
+    remoteFetchCb: RemoteFetchCb | undefined,
+) {
+    if (db.keyValueStoreRemote) {
+        const remoteChunk = await db.keyValueStoreRemote.read(key);
+        if (remoteChunk) {
+            if (remoteFetchCb) {
+                remoteFetchCb(key, remoteChunk);
+            }
+            // Return the remote chunk if it exists
+            return remoteChunk as T;
+        }
+    }
+
+    const localChunk = await db.keyValueStoreLocal.read(key);
```

Comment:
could duplication can be reduced by having two functions readChunkFromLocal and readChunkFromRemote that are beeing called in remoteFirstRead and localFirstRead